setup.py

```diff
@@ -47,7 +47,7 @@ def run(self):
 
 if platform.system() == "Windows":
     extra_compile_args.extend([
-        '-std=c++11'
+        '/std:c++14'
```

Comment:
Why would Windows need a higher C++ version (C++14) than Mac or Linux? If C++11 doesn’t work for MSVC then maybe leave a code comment with the reason.

Reply:
From the MSVC [documentation](https://docs.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version?view=vs-2019) I understood that the minimal available C++ standard was the `14`. But I'm really not an expert on Microsoft compiler, the change isn't maybe necessary ?

Reply:
Aha. Apparently C++11 is [enabled by default](https://stackoverflow.com/questions/47043869/is-c11-available-in-visual-studio-2017/47043985) in Visual Studio 17. I did some further reading and I think your solution of the C++14 flag is correct. [There is no C++11 switch.](https://devblogs.microsoft.com/cppblog/standards-version-switches-in-the-compiler/)

Another helpful point of reference is Python’s compilers, which should be supported to the extent possible. [Python is built with “Visual C++ 14.X.”](https://wiki.python.org/moin/WindowsCompilers#Which_Microsoft_Visual_C.2B-.2B-_compiler_to_use_with_a_specific_Python_version_.3F) Visual Studio 2015 is Visual C++ 14.0, Visual Studio 2017 is 14.1, and Visual Studio 2019 is 14.2. It seems like this supports 2017 or newer, which is when standards compliance was improved [and the `/std` flag was added](https://docs.microsoft.com/en-us/cpp/build/reference/std-specify-language-standard-version). I think this solution is good. Perhaps add this code comment:
```suggestion
        # Supported by Visual C++ >=14.1
        '/std:c++14'
```